api/routes/posts.js

```diff
@@ -1,130 +1,131 @@
+      res.status(200).send(data);
+    })
+    .catch((err) => {
+      console.log(err);
+      res.sendStatus(400);
+    });
 });
 
 //to get postes with a city name
-app.get('/posts/:city', (req, res) => {
-	const city = req.params;
+app.get("/posts/:city", authenticated, (req, res) => {
+  const city = req.params;
 
-	Post.findAll({
-		where: { city: city.city },
-		attributes: ['topic', 'discription', 'city'],
-		include: [
-			{
-				model: User,
-				as: 'user',
-				attributes: ['firstName', 'lastName', 'email'],
-			},
-		],
-	})
-		.then((data) => {
-			console.log(data);
-			res.status(200).send('data');
-		})
-		.catch((err) => {
-			console.log(err);
-			res.status(400);
-		});
+  Post.findAll({
+    where: { city: city.city },
+    attributes: ["topic", "discription", "city"],
+    include: [
+      {
+        model: User,
+        as: "user",
+        attributes: ["firstName", "lastName", "email"],
+      },
+    ],
+  })
+    .then((data) => {
+      console.log(data);
+      res.status(200).send("data");
+    })
+    .catch((err) => {
+      console.log(err);
+      res.status(400);
+    });
 });
 
-///to create a post
-app.post('/posts/:id', (req, res) => {
-	const user = req.params;
-	if (!req.body.topic) {
-		res.status(400).send({
-			message: 'Content can not be empty!',
-		});
-		return;
-	}
+///Creates a new post for a user
+app.post("/posts/:id", authenticated, (req, res) => {
+  const user = req.params;
+  if (!req.body.topic) {
+    res.status(400).send({
+      message: "Content can not be empty!",
+    });
+    return;
+  }
 
-	const post = {
-		topic: req.body.topic,
-		city: req.body.city,
-		discription: req.body.discription,
-		published: req.body.published ? req.body.published : false,
-		user_id: user.id,
-	};
-	Post.create(post)
-		.then((data) => {
-			console.log(data);
-			res.status(201).send(data);
-		})
-		.catch((err) => {
-			res.status(500).send({
-				message: err.message || 'Some error occurred while creating the Post.',
-			});
-		});
+  const post = {
+    topic: req.body.topic,
+    post_city: req.body.post_city,
+    description: req.body.discription,
+    published: req.body.published ? req.body.published : false,
+    user_id: user.id,
+  };
+  Post.create(post)
+    .then((data) => {
+      console.log(data);
+      res.status(201).send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while creating the Post.",
+      });
+    });
 });
-app.delete('/posts/:id', (req, res) => {
-	const user = req.params.id;
-	console.log(user);
-	Post.destroy({
-		where: {
-			id: user,
-		},
-	})
-		.then((data) => {
-			console.log(data);
-			res.status(200).send('Post is deleted!');
-		})
-		.catch((err) => {
-			console.log(err);
-			res.status(400);
-		});
+// Deletes a post
+app.delete("/posts/:id", (req, res) => {
+  const user = req.params.id;
+  //console.log(user);
+  Post.destroy({
+    where: {
+      id: user,
```

Comment:
here instead of id it should be post_id as defined in the model. Otherwise is not working for me

Reply:
@gelilaa , Can you place here also` authenticated` middleware, This route will remove all posts belong to this user? Are you sure about this route. I think that maybe I would just remove one post.?

Reply:
cant find "destroyOne"Sequelize so i made it to find one post by id